spark/v3.4/spark/src/main/java/org/apache/iceberg/spark/SparkTableUtil.java

```diff
@@ -403,12 +403,12 @@ public static void importSparkTable(
       } else {
         List<SparkPartition> sourceTablePartitions =
             getPartitions(spark, sourceTableIdent, partitionFilter);
-        Preconditions.checkArgument(
-            !sourceTablePartitions.isEmpty(),
-            "Cannot find any partitions in table %s",
-            sourceTableIdent);
-        importSparkPartitions(
-            spark, sourceTablePartitions, targetTable, spec, stagingDir, checkDuplicateFiles);
+        if (sourceTablePartitions.isEmpty()) {
+          targetTable.newAppend().commit();
```

Comment:
What's the reason of empty commit, instead of say, returning ?

Reply:
That's what we do for un-partitioned table. And we expect the migration produces a snapshot so that we know the number of migrated files: https://github.com/apache/iceberg/blob/master/spark/v3.4/spark/src/main/java/org/apache/iceberg/spark/actions/MigrateTableSparkAction.java#L159